torch/_dynamo/variables/builtin.py

```diff
@@ -1160,6 +1160,15 @@ def call_getattr(
             else:
                 return VariableBuilder(tx, source)(member).add_guards(guards)
         elif isinstance(obj, (PythonModuleVariable, DummyModule)):
+            # np.typecodes -> tnp.typecodes, both module-level Dict[Str, Str]
+            import numpy as np
+            import torch._numpy as tnp
+
+            if obj.value == np and name == "typecodes":
+                tcodes = tnp.typecodes
+                result = {k: ConstantVariable.create(v) for k, v in tcodes.items()}
+                return ConstDictVariable(result, user_cls=dict)
```

Comment:
We can be a bit more general, and it will save us quite some pain in the future. We can do the np -> tnp swap, and just pull the element from tnp here. We would need to change a little bit the code and break here if the attribute does not exist, but I think that it makes a lot of sense.

We can land this one after we land the tests with dynamo one, so that we make sure we first get a good NumPy coverage with the tests.